src/internal/connector/graph/middleware.go

```diff
@@ -181,6 +182,40 @@ type RetryHandler struct {
 	Delay time.Duration
 }
 
+// Intercept implements the interface and evaluates whether to retry a failed request.
+func (middleware RetryHandler) Intercept(
+	pipeline khttp.Pipeline,
+	middlewareIndex int,
+	req *http.Request,
+) (*http.Response, error) {
+	ctx := req.Context()
+
+	response, err := pipeline.Next(req, middlewareIndex)
+	if err != nil && !IsErrTimeout(err) {
+		return response, stackReq(ctx, req, response, err)
+	}
+
+	exponentialBackOff := backoff.NewExponentialBackOff()
+	exponentialBackOff.InitialInterval = middleware.Delay
```

Comment:
Curious, what is the maximum possible exponential backoff period in our setup?

Reply:
Whatever we configure it to.  Default is `3`.